thefuck/shells.py

```diff
@@ -127,15 +127,15 @@ def _get_history_line(self, command_script):
 shells = defaultdict(lambda: Generic(), {
     'bash': Bash(),
     'zsh': Zsh(),
-    '-csh': Tcsh(),
+    'csh': Tcsh(),
     'tcsh': Tcsh()})
 
 
 def _get_shell():
     try:
-        shell = Process(os.getpid()).parent().cmdline()[0]
+        shell = Process(os.getpid()).parent().name()
     except TypeError:
-        shell = Process(os.getpid()).parent.cmdline[0]
+        shell = Process(os.getpid()).parent.name()
```

Comment:
BTW, should this be `Process(os.getpid()).name` if parent is None?

Since the version of psutil is not specified I don't know if I should change this line.

[Process.parent](https://pythonhosted.org/psutil/_modules/psutil.html#Process.parent)